main.py

```diff
@@ -1,7 +1,38 @@
 #!/usr/bin/env python3
+# This is the Hello-World-Python program ported to Python3 and improved
+# by using the proper enterprise project patterns.
+# Originally written by kmachnicki
+# Disclaimer: Yes, it is a joke :)
 
-#Welcome!
 
-print("Hello World!")
+# This is the heart of the program.
+# This class is responsible for printing the string provided by the caller.
+# It is a proxy for the print function.
+class Printer:
+    # __init__ is responsible for initializing the _printer member, so we can use it later.

```

Comment:
Class/Functions/Methods descriptions should happen using the triple quote multiline comment. Please apply to all functions.